File: Basics_with_JavaScript/1314 Nested Loops/14 Nested loops Excercise/03. Sum Prime Non Prime.js
function vladi(array) {
  primeSum = 0;
  nonPrimeSum = 0;
  let i = 0;
  let command = array[i];
  i++;

  while (command !== "stop") {
    let currentNumber = Number(command);
    let isPrime = true;

    if (currentNumber < 0) {
      console.log(`Number is negative.`);
    } else if (currentNumber === 1) {
      isPrime = false;
    } else {
      for (let j = 2; j <= currentNumber - 1; j++) {
        if (currentNumber % j === 0) {
          isPrime = false;
          break;
        }
      }
    }

    if (isPrime && currentNumber > 0) {
      primeSum += currentNumber;
    } else if (!isPrime && currentNumber > 0) {
      nonPrimeSum += currentNumber;
    }

    command = array[i];
    i++;
  }

  console.log(`Sum of all prime numbers is: ${primeSum}`);
  console.log(`Sum of all non prime numbers is: ${nonPrimeSum}`);
}

vladi(["3", "9", "0", "7", "19", "4", "stop"]);
vladi(["30", "83", "33", "-1", "20", "stop"]);
vladi(["0", "-9", "0", "stop"]);

File: Basics_with_Python/PythonBasicCourse/6_Nested Loops/14_Nested Loops - Exercise/03. Sum Prime Non Prime.py
number = input()
prime_sum = 0
not_prime_sum = 0
while number != "stop":
    number = int(number)
    if number < 0:
        print("Number is negative.")
        number = input()
        continue
    else:
        prime = True
        for check in range(2, number):
            if number % check == 0:
                prime = False
                break
        if prime:
            prime_sum += number
        else:
            not_prime_sum += number
        number = input()
print(f"Sum of all prime numbers is: {prime_sum}")
print(f"Sum of all non prime numbers is: {not_prime_sum}")
